src/python/pants/option/options.py

```diff
@@ -65,6 +65,15 @@ class Options:
       - None.
     """
 
+    # This is a class-wide toggle for whether Python warnings (using the warnings module) should
+    # be emitted during the options-parsing process. We want to initially set this to `True` and
+    # then set it to `False` after we have done the initial options parsing.
+    _suppress_warnings = True
```

Comment:
So, there is a particular place where the bootstrap options are always parsed: https://github.com/pantsbuild/pants/blob/74eda8542bb84b551a1aa8a54a4d7ea69af46329/src/python/pants/option/options_bootstrapper.py#L80-L96 ... which is called by OptionsBootstrapper.create.

It feels like it would be a bit cleaner to wrap the bootstrap options parse inside OptionsBootstrapper.create in https://docs.python.org/3/library/warnings.html#warnings.catch_warnings , and then rely on callers of `OptionsBootstrapper.create` to install the warnings filters immediately afterward (where they presumably set log levels right now?)

Reply:
We also call `parse_bootstrap_options` in the `bootstrap_options` memoized property on that class, which we use in a number of places. In any case, using that `catch_warnings` function in `parse_bootstrap_options` doesn't seem to actually silence the warnings, and I"m not sure why.